tests/actor/test_power_distributing.py

```diff
@@ -738,3 +740,43 @@ async def test_power_distributor_stale_all_components_message(
         assert result.status == Result.Status.ERROR
         # User is interested in batteries only.
         assert result.error_message == "No data for the given batteries {106, 206}"
+
+
+class TestBrokenComponents:
+    """Test if BrokenComponents class is working as expected."""
+
+    def test_broken_components(self, mocker: MockerFixture) -> None:
+        """Check if components are blocked for 30 seconds.
+
+        Args:
+            mocker (MockerFixture): pytest mocker
+        """
+        datetime_mock = mocker.patch("frequenz.sdk.actor.power_distributing.datetime")
+
+        expected_datetime = [
+            datetime.fromisoformat("2001-01-01T00:00:00+00:00"),
+            datetime.fromisoformat("2001-01-01T00:00:10+00:00"),
+            datetime.fromisoformat("2001-01-01T00:00:20+00:00"),
+        ]
+        expected_datetime.extend(
+            20 * [datetime.fromisoformat("2001-01-01T00:00:31+00:00")]
+        )
+
+        datetime_mock.now.side_effect = expected_datetime
+
+        # After 30 seconds components should be considered as working
+        broken = _BrokenComponents(30)
+
+        for i in range(3):
+            broken.mark_as_broken(i)
```

Comment:
interesting,  pylint always complains to me when I use single letter variable names.

Reply:
Done

Reply:
Maybe it is related with python version number?
What python you are using?

Reply:
it is fine,  I was just worried that pylint might not be running on the test files for some reason and we are not catching other issues as well.  I will test it separately.

Reply:
it probably have an exception for `for` loops.